File: feature/src/main/java/org/indiv/dls/games/verboscruzados/feature/dialog/StatsDialogFragment.java
package org.indiv.dls.games.verboscruzados.feature.dialog;

import org.indiv.dls.games.verboscruzados.feature.R;

import android.app.Dialog;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

public class StatsDialogFragment extends DialogFragment {

    private int mGamesCompleted;
    private int mWordsCompleted;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.fragment_stats_dialog, null);

        // use the Builder class for convenient dialog construction
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity())
                .setPositiveButton(R.string.dialog_ok, (dialog, id) -> { })
                .setView(view);

        // create the AlertDialog object 
        Dialog dialog = builder.create();

        // fill in stats
        appendTextToView(view.findViewById(R.id.textview_gamescompleted), " " + mGamesCompleted);
        appendTextToView(view.findViewById(R.id.textview_wordscompleted), " " + mWordsCompleted);

        return dialog;
    }


    //---------------------------------------------------//
    //------------ end of overridden methods ------------//
    //---------------------------------------------------//


    public void setStats(int gamesCompleted, int wordsCompleted) {
        mGamesCompleted = gamesCompleted;
        mWordsCompleted = wordsCompleted;
    }

    private void appendTextToView(TextView textView, String text) {
        textView.setText(textView.getText() + text);
    }
}

File: feature/src/main/java/org/indiv/dls/games/verboscruzados/feature/dialog/StatsDialogFragment.kt
package org.indiv.dls.games.verboscruzados.feature.dialog

import org.indiv.dls.games.verboscruzados.feature.R

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import android.widget.TextView
import org.indiv.dls.games.verboscruzados.feature.game.PersistenceHelper
import org.indiv.dls.games.verboscruzados.feature.model.ConjugationType
import org.indiv.dls.games.verboscruzados.feature.model.InfinitiveEnding
import org.indiv.dls.games.verboscruzados.feature.model.IrregularityCategory

/**
 * Dialog for showing game stats.
 */
class StatsDialogFragment : DialogFragment() {

    //region COMPANION OBJECT ----------------------------------------------------------------------

    companion object {
        /**
         * Creates stats index for 2-dimensional representation of stats where IrregularityCategory
         * and InfinitiveEnding are on the y-axis, and conjugation type on the x-axis.
         */
        fun createStatsIndex(conjugationType: ConjugationType,
                             infinitiveEnding: InfinitiveEnding,
                             irregularityCategory: IrregularityCategory): Int {
            val rowIndex = (irregularityCategory.indexForStats * 3) + infinitiveEnding.indexForStats
            return rowIndex * ConjugationType.values().size + conjugationType.indexForStats
        }
    }

    //endregion

    //region PUBLIC PROPERTIES ---------------------------------------------------------------------
    //endregion

    //region PRIVATE PROPERTIES --------------------------------------------------------------------
    //endregion

    //region PUBLIC INTERFACES ---------------------------------------------------------------------
    //endregion

    //region OVERRIDDEN FUNCTIONS ------------------------------------------------------------------

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val inflater = activity!!.layoutInflater
        val v = inflater.inflate(R.layout.fragment_stats_dialog, null)

        val dialog = AlertDialog.Builder(activity!!)
                .setTitle(R.string.dialog_stats_heading)
                .setPositiveButton(R.string.dialog_ok) { dialog, id -> }
                .setView(v)
                .create()

        // fill in stats
        val persistenceHelper = PersistenceHelper(activity!!)
        val statsMap = persistenceHelper.allGameStats

//        val textView: TextView = v.findViewById(R.id.textview_temp)
        val myText = statsMap.toString()
//        textView.text = myText

        return dialog
    }

    //endregion

    //region PUBLIC FUNCTIONS ----------------------------------------------------------------------
    //endregion

    //region PRIVATE FUNCTIONS ---------------------------------------------------------------------
    //endregion
}
